src/components/Anchor/Anchor.js

```diff
@@ -170,8 +170,6 @@ class Anchor extends React.Component {
 
   handleClick = event => {
     const { onClick, to, disabled } = this.props;
-    // Don't handle clicks if disabled
-    if (disabled) return;
```

Comment:
To make disabled function correctly, I think we should go with your original proposal... return early while calling `event.preventDefault();`.

That makes it so that a user can replace the entire Anchor rendering component and still have disabled work.